### PR TITLE
Little changes to Account and AccountLine classes

### PR DESCRIPTION
**Pull request by marcosgdf**
Changes are detailed in separated commits to allow an easy review

### CODE REVIEW

File: htdocs/compta/bank/ligne.php [hunk 1]
```diff
@@ -60,16 +60,16 @@
  * Actions
  */
 
-if ($user->rights->banque->consolidate && $action == 'dvnext')
+if ($user->rights->banque->consolidate)
```
Comment (eldy) [inline htdocs/compta/bank/ligne.php]: If we call the page with no action or any action, we will execute this code when we should not, wouldn't we ?
We should add test if $action start with dv for example.

File: htdocs/compta/bank/releve.php [hunk 2]
```diff
@@ -52,19 +52,18 @@
 if ($user->societe_id) $socid=$user->societe_id;
 $result=restrictedArea($user,'banque',$fieldid,'bank_account','','',$fieldname);
 
-if ($user->rights->banque->consolidate && $action == 'dvnext' && ! empty($dvid))
+if ($user->rights->banque->consolidate && ! empty($dvid))
```
Comment (eldy) [inline htdocs/compta/bank/releve.php:55]: If we call the page with no action or any action, we will execute this code when we should not, wouldn't we ?
We should add test if $action start with dv for example.

**Comment by eldy** [comment]
The code seems to generate a lot of fetch, tracks into logs and adding new object usage, just to insert a record into a table that is not an object table but a relation/association table.
What is the benefit compared to old code (less code to do the same) ?

**Comment by marcosgdf** [comment]
I thought that was considered an actual advantage over the old way of working.

Having centralized data management allows us to keep control of the code and the data. Also it is not a relation/association table but an object table.

**Comment by eldy** [comment]
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolte, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.

**Pull request closed without merging**